extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/AbstractWebSocketEndpointWrapper.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import java.lang.reflect.Type;
+
+import io.quarkus.websockets.next.InboundProcessingMode;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * {@link WebSocketEndpoint} wrapper that delegates all methods to {@link #delegate}.
+ * This way, subclasses can only override methods they need to intercept.
+ */
+abstract class AbstractWebSocketEndpointWrapper implements WebSocketEndpoint {
```

Comment:
Given the functionality this class provides it should be probably named `ForwardingWebSocketEndpoint`.